core/services/ocr2/plugins/directrequestocr/reporting.go

```diff
@@ -99,6 +139,7 @@ func (r *functionsReporting) Query(ctx context.Context, ts types.ReportTimestamp
 		"queryLen":   len(queryProto.RequestIDs),
 		"requestIDs": idStrs,
 	})
+	promReportingPluginsQuery.WithLabelValues(r.jobID.String()).Inc()
```

Comment:
Can those calls be blocking? All phases have Contexts with a timeout.

Reply:
This does not perform any IO. It just increments a counter in-memory.